Assets/Scripts/State/IdleState.cs

```diff
@@ -26,11 +29,41 @@ public IdleState(Character character, State nextState = null)
 
         public override void Update(float deltaTime)
         {
+                Tile startTile = character.CurrTile;
+                Tile endTile = null;
+                List<Tile> path = null;
+
+                // Get a random Int 0,1,2,3 for representing the 4 directions.
+                int direction = Int32.Parse(Math.Truncate(Random.value * 4).ToString());
+
+                switch (direction)
+                {
+                    case 0:
+                        endTile = World.Current.GetTileAt(startTile.X + 1, startTile.Y, startTile.Z);
+                        break;
+                    case 1:
+                        endTile = World.Current.GetTileAt(startTile.X, startTile.Y + 1, startTile.Z);
+                        break;
+                    case 2:
+                        endTile = World.Current.GetTileAt(startTile.X - 1, startTile.Y, startTile.Z);
+                        break;
+                    case 3:
+                        endTile = World.Current.GetTileAt(startTile.X, startTile.Y - 1, startTile.Z);
+                        break;
+                }
+
+                if (endTile != null)
+                {
+                  path = Pathfinder.FindPathToTile(startTile, endTile);
+                }
+
+            Pathfinder.GoalEvaluator GTE = Pathfinder.GoalTileEvaluator(endTile, true);
+
             timeSpentIdle += deltaTime;
             if (timeSpentIdle >= totalIdleTime)
             {
-                // We are done. Lets look for work.
-                character.SetState(null);
+                // We are done. Execute move then look for work.
+              character.SetState(new MoveState(character, GTE, path));
```

Comment:
GTE is a bad variable name, it tells us nothing about what it contains. Also you could easily do `character.SetState(new MoveState(character, Pathfinder.GoalTileEvaluator(endTile, true), path));` which is much more direct and clear as to what is being passed (since it isn't used for anything but passing to SetState), avoids the non-informational variable name, and is also how GoalEvaluators are passed in every other place in code (that I am aware of).